Sources/AppCoinsSDK/Models/Services/PayFlow/PayFlowClient.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  PayFlowClient.swift
+//  
+//
+//  Created by Graciano Caldeira on 25/07/2024.
+//
+
+import Foundation
+
+internal class PayFlowClient: PayFlowService {
+    
+    private let endpoint: String
+    
+    init(endpoint: String = BuildConfiguration.payFlowServiceBaseURL) {
+        self.endpoint = endpoint
+    }
+    
+    internal func setPayFlow(package: String, packageVercode: String, sdkVercode: Int, locale: String?, oemID: String?, oemIDType: String?, country: String?, os: String, result: @escaping (Result<PayFlowDataRaw, PayFlowError>) -> Void) {
+        if let requestUrl = URL(string: "\(endpoint)/api/v2/payment_flow") {
+            
+            var queryItems = [
+                URLQueryItem(name: "package", value: package),
+                URLQueryItem(name: "package_vercode", value: packageVercode),
+                URLQueryItem(name: "sdk_vercode", value: String(sdkVercode)),
+                URLQueryItem(name: "locale", value: locale),
+                URLQueryItem(name: "oemid", value: oemID),
+                URLQueryItem(name: "oemid_type", value: oemIDType),
+                URLQueryItem(name: "country", value: country),
+                URLQueryItem(name: "os", value: os)
+            ]
+            
+            queryItems = queryItems.filter { $0.value != nil }
+            var components = URLComponents(url: requestUrl, resolvingAgainstBaseURL: false)
+            components?.queryItems = queryItems
+            
+            guard let url = components?.url else { fatalError() }
```

Comment:
We don't use `fatalError()`, you need to solve it as if the request failed but this ain't it.